### ISSUE TITLE
How to output the enhanced data from the training process?

### ISSUE THREAD

**Issue by gml-blip** [ultralytics/ultralytics#17209]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I used the following code for training：
from ultralytics import YOLO
import os
# Load a model
model = YOLO("yolov8n.yaml")  # build a new model from YAML
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)
model = YOLO("yolov8n.yaml").load("yolov8n.pt")  # build from YAML and transfer weights

# Train the model

results = model.train(data="datasets/data/mydata7/mydata.yaml",
                      hsv_s=0.1,
                      hsv_v=0.6,
                      degrees=10,
                      scale=0.7,
                      shear=5,
                      fliplr=1,
                      mixup=0.3,
                      epochs=100,
                      imgsz=640)
 How to output the enhanced data from the training process?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @gml-blip, thank you for your interest in Ultralytics 🚀! Your question about outputting enhanced data from the training process is a great one.

For guidance on data augmentation and how to handle outputs, we recommend reviewing our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/). Additionally, ensure you've seen the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) documentation for more examples which might provide further insights.

If this is a 🐛 Bug Report, please make sure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate further. For questions like these, it often helps to share dataset image examples and the specific configurations you're using.

Meanwhile, you might want to join our vibrant community for more interactive discussions. Feel free to engage with other users on [Discord](https://ultralytics.com/discord) 🎧, participate in broader discussions on [Discourse](https://community.ultralytics.com), or exchange ideas with fellow practitioners on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Please ensure your `ultralytics` package is up-to-date, as enhancements and fixes are continually integrated. You can upgrade with:

```bash
pip install -U ultralytics
```

Ensure you're working with [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/), which might resolve some issues.

## Environments

YOLO operates seamlessly in the following verified environments, already set up with necessary dependencies:

- **Notebooks** with GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

For the moment, this is an automated response 😊, but rest assured that an Ultralytics engineer will review your issue and assist you soon. If the CI badge is green, it means all current tests are passing, ensuring the robustness of our tools.

Thanks for reaching out and happy experimenting with Ultralytics!

**Comment by Y-T-G** [comment]
You can load the `best.pt` file in the `runs/detect/train/weights` folder and use it to predict.

https://docs.ultralytics.com/modes/predict/

**Comment by gml-blip** [comment]
> You can load the `best.pt` file in the `runs/detect/train/weights` folder and use it to predict.您可以在 `runs/detect/train/weights` 文件夹中加载 `best.pt` 文件，并使用它来预测。
> 
> https://docs.ultralytics.com/modes/predict/

Thanks for your answer, but I would like to make an output of the dataset enhanced by the data during training, that is, an output of the changes in the dataset after setting the hyperparameters.

**Comment by Y-T-G** [comment]
You mean augmentations?

**Comment by gml-blip** [comment]
> You mean augmentations?

yes

**Comment by Y-T-G** [comment]
There isn't a method to save the augmentations. It's fine on the fly.
